snap/hooks/prepare-plug-plugin

```diff
@@ -4,11 +4,13 @@
 # Prefer an up-to-date snapctl from the core or snapd snaps, if they exist. We ask users to install
 # the core snap in the Certbot installation instructions.
 # See https://github.com/certbot/certbot/issues/8922, https://bugs.launchpad.net/snapd/+bug/1933392
+SNAPCTL_CORE="/snap/core/current/usr/bin/snapctl"
+SNAPCTL_SNAPD="/snap/snapd/current/usr/bin/snapctl"
 SNAPCTL="snapctl"
-if [ -x "/snap/core/current/usr/bin/snapctl" ]; then
-    SNAPCTL="/snap/core/current/usr/bin/snapctl"
-elif [ -x "/snap/snapd/current/usr/bin/snapctl" ]; then
-    SNAPCTL="/snap/snapd/current/usr/bin/snapctl"
+if $SNAPCTL_CORE get x 2>/dev/null; then
```

Comment:
How did you test this? When I run `snap get certbot x` on the command line, it exits 1 because we haven't set key `x`. Is it documented somewhere that `snapctl` has different behavior?

Reply:
> How did you test this? 

One way:

    $ sudo snap run --shell certbot
    root@x1:~/devel/certbot# snapctl get x

    root@x1:~/devel/certbot# echo $?
    0

But mainly I tested it by trying out the resulting snap.

>Is it documented somewhere that snapctl has different behavior?

No, I didn't find any documentation on this one way or the other.

https://snapcraft.io/docs/using-snapctl does mention that it uses exit codes for the `is-connected` subcommand, but nothing for `get`/`set`.

Do you have another idea in mind for probing the functionality of `snapctl`? If not, I could post on the snapcraft forum and ask about this.

Reply:
No, I like it, I was just worried by the disparity with `snap get`. Very weird that they'd differ; thanks for the explanation, lgtm.

Reply:
Actually, now that I've said that, I would prefer using documented behavior, which doesn't seem to exist. If we post on the forums to ask, perhaps that answer would be canonical and I'd feel comfortable relying on it. But that being said, I think I'd rather merge this to get the fix in for tomorrow's release, and ask later and change it if we need to.

Reply:
Asked at https://forum.snapcraft.io/t/is-the-exit-code-0-of-snapctl-get-some-nonexistent-option-stable/25841.

Reply:
There has been a response: https://forum.snapcraft.io/t/is-the-exit-code-0-of-snapctl-get-some-nonexistent-option-stable/25841/3.